File: 2614-prime-in-diagonal/main.go
func diagonalPrime(nums [][]int) int {

    if len(nums) == 0 {
        return 0
    }

    var primeNumbers []int
    for i := range nums {
        primeNumbers = append(primeNumbers, nums[i][i])
        primeNumbers = append(primeNumbers, nums[i][len(nums) - i - 1])
    }

    return maxPrime(primeNumbers)
}

func maxPrime(primeNumbers []int) int {
    maxPrime := 0
    for _, num := range primeNumbers {
        if isPrime(num) && num > maxPrime {
            maxPrime = num
        }
    }

    return maxPrime
}

func isPrime(num int) bool {
    
    if num <= 1 {
        return false
    }

    for i := 2; i < int(math.Pow(float64(num), 0.5)) + 1; i++ {
        if num % i == 0 {
            return false
        }
    }

    return true
}

File: 2614-prime-in-diagonal/main.rs
impl Solution {
    pub fn diagonal_prime(nums: Vec<Vec<i32>>) -> i32 {
        
        if nums.is_empty() {
            return 0;
        }

        let mut prime_numbers: Vec<i32> = Vec::with_capacity(nums.len() * 2);
        for i in 0..nums.len(){
            prime_numbers.push(nums[i][i]);
            prime_numbers.push(nums[i][nums.len() - i - 1]);
        }

        let mut max_prime = 0;
        for num in prime_numbers {
            if is_prime(num) && num > max_prime {
                max_prime = num;
            }
        }

        max_prime
    }
}

fn is_prime(num: i32) -> bool {
    if num <= 1 {
        return false;
    }

    for i in 2..=((num as f64).sqrt() as i32) {
        if num % i == 0 {
            return false;
        }
    }

    true
}
